server/sockets/cohort.js

```diff
@@ -237,218 +262,139 @@ const updateCohortHeaderStatus = (socket, cohortClientLocks) => {
-        const cohort = responseWithCohort(doc);
-        memberIds.forEach(id => {
-          const memberId = id.toString();
-          if (allCohortsClients.has(memberId)) {
-            const { socketId } = allCohortsClients.get(memberId);
-            io.sockets
-              .to(socketId)
-              .emit(CohortActionTypes.DELETE_SUCCESS, { cohortId });
-          }
+    const listIds = lists.map(list => list._id.toString());
+
+    listIds.forEach(id => {
+      const listId = id.toString();
+
+      io.sockets
+        .to(listChannel(listId))
+        .emit(CohortActionTypes.ARCHIVE_SUCCESS, {
+          listId,
+          performer
         });
-        ownerIds.forEach(id => {
-          const ownerId = id.toString();
-          if (allCohortsClients.has(ownerId)) {
-            const { socketId } = allCohortsClients.get(ownerId);
-            io.sockets
-              .to(socketId)
-              .emit(CohortActionTypes.FETCH_META_DATA_SUCCESS, {
-                [cohortId]: cohort
-              });
-          }
+
+      io.sockets
+        .to(listMetaDataChannel(listId))
+        .emit(CohortActionTypes.ARCHIVE_SUCCESS, {
+          listId
         });
-      }
     });
+  } catch {
+    // Ignore errors
+  }
+
+  return Promise.resolve();
 };
 
-const removeMember = (
-  io,
-  allCohortsClients,
-  cohortClients,
-  dashboardClients,
-  listClients
-) => data => {
-  const { cohortId, userId } = data;
-  const removedUserId = userId.toString();
+const removeMember = io => async data => {
+  const { cohortId, membersCount, performer, userId } = data;
 
   io.sockets
     .to(cohortChannel(cohortId))
-    .emit(CohortActionTypes.REMOVE_MEMBER_SUCCESS, data);
-
-  emitCohortMetaData(cohortId, allCohortsClients, io);
-
-  if (cohortClients.has(removedUserId)) {
-    const { viewId, socketId } = cohortClients.get(removedUserId);
-
-    if (viewId === cohortId) {
-      io.sockets.to(socketId).emit(CohortActionTypes.REMOVED_BY_SOMEONE, {
-        cohortId
-      });
-    }
-  }
+    .emit(CohortActionTypes.REMOVE_MEMBER_SUCCESS, {
+      cohortId,
+      performer,
+      userId
+    });
 
-  if (allCohortsClients.has(removedUserId)) {
-    const { socketId } = allCohortsClients.get(removedUserId);
+  io.sockets
+    .to(cohortMetaDataChannel(cohortId))
+    .emit(CohortActionTypes.REMOVE_MEMBER_SUCCESS, { cohortId, userId });
 
-    io.sockets
-      .to(socketId)
-      .emit(CohortActionTypes.DELETE_SUCCESS, { cohortId });
-  }
+  io.sockets
+    .to(cohortMetaDataChannel(cohortId))
+    .emit(CohortActionTypes.UPDATE_SUCCESS, { cohortId, membersCount });
 
-  List.find({ cohortId })
-    .populate('cohortId', 'memberIds')
-    .lean()
-    .exec()
-    .then(docs => {
-      if (docs && docs.length > 0) {
-        const {
-          cohortId: { memberIds: cohortMemberIds }
-        } = docs[0];
-        const listIdsUserWasRemovedFrom = [];
-        const listIdsUserRemained = [];
+  try {
+    const listIdsUserRemained = [];
```

Comment:
what does this const means?

Reply:
if we remove the cohort user he has to be removed from shared lists only when he is a viewer only.  So this array holds lists' ids tha user was no removed from (he still remains there. The second array hold lists' ids that user war removed from.